### PR TITLE
feat(marxan-run): serve zipped output files

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/libs/marxan-output/src/marxan-execution-metadata.geo.entity.ts [hunk 1]
```diff
@@ -32,4 +38,7 @@ export class MarxanExecutionMetadataGeoEntity {
     name: `output_zip`,
   })
   outputZip?: Buffer | null;
+
+  @CreateDateColumn({ type: 'timestamp without time zone' })
```
Comment (hotzevzl) [inline api/libs/marxan-output/src/marxan-execution-metadata.geo.entity.ts:42]: can we please stick to `snake_case` for column names, even at the expense of explicitly setting `name: 'created_at'`?

**Review by hotzevzl — APPROVED** [review]

File: api/libs/marxan-output/src/marxan-execution-metadata.geo.entity.ts (hunk 1 above)
Comment (kgajowy) [reply api/libs/marxan-output/src/marxan-execution-metadata.geo.entity.ts:42]: good catch, thanks!

File: api/apps/api/test/scenario-run-results/world.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      await ScenariosTestUtils.deleteScenario(app, token, scenario.id);
+      await cleanup();
+    },
+    GivenOutputZipIsAvailable: async () => {
+      const zipBuffer = await createArchive();
+      filesToRemove.push(...zipBuffer.filesToClean);
+      await marxanExecutionMetadataRepo.save(
+        marxanExecutionMetadataRepo.create({
+          outputZip: zipBuffer.archive,
+          inputZip: zipBuffer.archive,
```
Comment (Dyostiq) [inline api/apps/api/test/scenario-run-results/world.ts:49]: if output and input is the same, you can't recognize which one is really served

File: api/apps/api/test/scenario-run-results/world.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      );
+      expect(response.header[`content-disposition`]).toEqual(
+        `attachment; filename="output.zip"`,
+      );
+      const directory = await unzipper.Open.buffer(response.body);
+      expect(directory.files.length).toEqual(1);
+      expect(directory.files[0].path).toEqual(`hello.txt`);
+      expect((await directory.files[0].buffer()).toString()).toEqual(
+        `Hello Marxan!`,
+      );
```
Comment (Dyostiq) [inline api/apps/api/test/scenario-run-results/world.ts:70]: wouldn't buffer comparison be enough?

**Review by Dyostiq — APPROVED** [review]

File: api/apps/api/test/scenario-run-results/world.ts (hunk 3 above)
Comment (kgajowy) [reply api/apps/api/test/scenario-run-results/world.ts:70]: would need to save buffer in fixtures/world - just made a "shortcut" 😓

Comment (Dyostiq) [reply api/apps/api/test/scenario-run-results/world.ts:70]: ok :D

File: api/apps/api/src/modules/scenarios/scenarios.controller.ts [hunk 4]
```diff
@@ -271,6 +271,25 @@ export class ScenariosController {
...
+  @Header('Content-Disposition', 'attachment; filename="output.zip"')
+  async getOutputArchive(
+    @Param(`id`, ParseUUIDPipe) scenarioId: string,
+    @Res() response: Response,
+  ) {
+    const file = await this.service.getMarxanExecutionOutputArchive(scenarioId);
+    response.set({
+      [`Content-Type`]: `application/octet-stream`,
+      [`Content-Disposition`]: `attachment; filename="output.zip"`,
+    });
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenarios/scenarios.controller.ts:289]: AFAIR it is not needed, despite having `Res` in the method, `Header` decorators still control headers.

Comment (kgajowy) [reply api/apps/api/src/modules/scenarios/scenarios.controller.ts]: https://github.com/Vizzuality/marxan-cloud/pull/319/commits/e9e74a6875780002b457deec5bc20df6c0cae77e

File: api/apps/api/test/scenario-run-results/world.ts (hunk 2 above)
Comment (kgajowy) [reply api/apps/api/test/scenario-run-results/world.ts]: https://github.com/Vizzuality/marxan-cloud/pull/319/commits/e9e74a6875780002b457deec5bc20df6c0cae77e

File: api/apps/api/src/modules/scenarios/scenarios.controller.ts [hunk 5]
```diff
@@ -271,6 +271,25 @@ export class ScenariosController {
     return await this.service.getSpecDatCsv(id);
   }
 
+  @ApiTags(marxanRunFiles)
+  @ApiOperation({
+    description: `Get archived output files`,
+  })
+  @Get(`:id/marxan/output`)
+  @Header(`Content-Type`, `application/octet-stream`)
```
[the remark this answers is not in the document]
Comment (kgajowy) [reply api/apps/api/src/modules/scenarios/scenarios.controller.ts]: https://github.com/Vizzuality/marxan-cloud/pull/319/commits/e9e74a6875780002b457deec5bc20df6c0cae77e

File: api/apps/api/src/modules/scenarios/output-files/output-files.service.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  async get(scenarioId: string): Promise<Buffer> {
+    const latest = await this.executionMetadataRepo.find({
+      where: {
+        scenarioId,
+      },
+      order: {
+        createdAt: 'DESC',
+      },
+      take: 1,
+    });
```
[the remark this answers is not in the document]
Comment (kgajowy) [reply api/apps/api/src/modules/scenarios/output-files/output-files.service.ts]: https://github.com/Vizzuality/marxan-cloud/pull/319/commits/e9e74a6875780002b457deec5bc20df6c0cae77e

File: api/apps/api/src/modules/scenarios/output-files/output-files.service.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    if (latest.length !== 1) {
+      throw new Error('Not found');
+    }
+    if (!latest[0].outputZip) {
+      throw new Error('Not available yet. Please execute Marxan first.');
+    }
```
[the remark this answers is not in the document]
Comment (kgajowy) [reply api/apps/api/src/modules/scenarios/output-files/output-files.service.ts]: @Dyostiq https://github.com/Vizzuality/marxan-cloud/pull/319/commits/fa496d41ed90c97d17e4b6ff97b96a97f3207298

**Pull request merged**